sweep/sweeper.go

```diff
@@ -753,180 +615,13 @@ func (s *UtxoSweeper) CreateSweepTx(inputs []Input, confTarget uint32,
-	}
-
-	hashCache := txscript.NewTxSigHashes(sweepTx)
-
-	// With all the inputs in place, use each output's unique witness
-	// function to generate the final witness required for spending.
-	addWitness := func(idx int, tso Input) error {
-		witness, err := tso.BuildWitness(
-			s.cfg.Signer, sweepTx, hashCache, idx,
-		)
-		if err != nil {
-			return err
-		}
-
-		sweepTx.TxIn[idx].Witness = witness
-
-		return nil
-	}
-
-	// Finally we'll attach a valid witness to each csv and cltv input
-	// within the sweeping transaction.
-	for i, input := range inputs {
-		if err := addWitness(i, input); err != nil {
-			return nil, err
-		}
-	}
-
-	return sweepTx, nil
-}
-
-func getInputWitnessSizeUpperBound(input Input) (int, error) {
-	switch input.WitnessType() {
-
-	// Outputs on a remote commitment transaction that pay directly
-	// to us.
-	case lnwallet.CommitmentNoDelay:
-		return lnwallet.P2WKHWitnessSize, nil
-
-	// Outputs on a past commitment transaction that pay directly
-	// to us.
-	case lnwallet.CommitmentTimeLock:
-		return lnwallet.ToLocalTimeoutWitnessSize, nil
-
-	// Outgoing second layer HTLC's that have confirmed within the
-	// chain, and the output they produced is now mature enough to
-	// sweep.
-	case lnwallet.HtlcOfferedTimeoutSecondLevel:
-		return lnwallet.ToLocalTimeoutWitnessSize, nil
-
-	// Incoming second layer HTLC's that have confirmed within the
-	// chain, and the output they produced is now mature enough to
-	// sweep.
-	case lnwallet.HtlcAcceptedSuccessSecondLevel:
-		return lnwallet.ToLocalTimeoutWitnessSize, nil
-
-	// An HTLC on the commitment transaction of the remote party,
-	// that has had its absolute timelock expire.
-	case lnwallet.HtlcOfferedRemoteTimeout:
-		return lnwallet.AcceptedHtlcTimeoutWitnessSize, nil
-
-	// An HTLC on the commitment transaction of the remote party,
-	// that can be swept with the preimage.
-	case lnwallet.HtlcAcceptedRemoteSuccess:
-		return lnwallet.OfferedHtlcSuccessWitnessSize, nil
-	}
-
-	return 0, fmt.Errorf("unexpected witness type: %v", input.WitnessType())
-}
-
-// getWeightEstimate returns a weight estimate for the given inputs.
-// Additionally, it returns counts for the number of csv and cltv inputs.
-func getWeightEstimate(inputs []Input) ([]Input, int64, int, int) {
-	// We initialize a weight estimator so we can accurately asses the
-	// amount of fees we need to pay for this sweep transaction.
-	//
-	// TODO(roasbeef): can be more intelligent about buffering outputs to
-	// be more efficient on-chain.
-	var weightEstimate lnwallet.TxWeightEstimator
-
-	// Our sweep transaction will pay to a single segwit p2wkh address,
-	// ensure it contributes to our weight estimate.
-	weightEstimate.AddP2WKHOutput()
-
-	// For each output, use its witness type to determine the estimate
-	// weight of its witness, and add it to the proper set of spendable
-	// outputs.
-	var (
-		sweepInputs         []Input
-		csvCount, cltvCount int
+	return s.cfg.TxGenerator.createSweepTx(
```

Comment:
Should we then remove the `CreateSweepTx` and allow callers to directly access the tx generator instead?

Reply:
I changed it a bit again. Store the dependencies (signer etc) in sweeper and removed the tx generator struct.